crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,219 @@
+
+    // Vim and vscode temporary files
+    if path
+        .extension()
+        .and_then(|ext| ext.to_str())
+        .map_or(false, |ext| {
+            IGNORED_EXTENSIONS
+                .iter()
+                .any(|&ignored| ext.eq_ignore_ascii_case(ignored))
+        })
+    {
+        return true;
+    }
+
+    // Vim temporary files
+    if file_name.ends_with('~') {
+        return true;
+    }
+
+    // Emacs temporary files
+    if file_name.starts_with('#') && file_name.ends_with('#') {
+        return true;
+    }
+
+    // Add more patterns for other editors as needed
+
+    false
+}
+
+impl Cmd {
+    pub async fn run(&mut self) -> Result<(), Error> {
+        let (tx, mut rx) = mpsc::channel::<Message>(100);
+        let rebuild_state = Arc::new(Mutex::new(false));
+        let workspace_root: &Path = self
+            .build_cmd
+            .manifest_path
+            .parent()
+            .unwrap_or_else(|| Path::new("."));
+        let env_toml_path = workspace_root.join("environments.toml");
+
+        let packages = self
+            .build_cmd
+            .list_packages()?
+            .into_iter()
+            .map(|package| PathBuf::from(package.manifest_path.parent().unwrap().as_str()))
+            .collect::<Vec<_>>();
+
+        let watcher = Watcher::new(&env_toml_path, &packages);
+
+        for package_path in watcher.packages.iter() {
+            eprintln!("Watching {}", package_path.display());
+        }
+
+        let watcher_clone = watcher.clone();
+        let mut notify_watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
+                    if matches!(
+                        event.kind,
+                        notify::EventKind::Create(_)
+                            | notify::EventKind::Modify(_)
+                            | notify::EventKind::Remove(_)
+                    ) {
+                        if let Some(path) = event.paths.first() {
+                            if is_temporary_file(path) {
+                                return;
+                            }
+                            if watcher_clone.is_watched(path) || watcher_clone.is_env_toml(path) {
+                                eprintln!("File changed: {path:?}");
+                                if let Err(e) = tx.blocking_send(Message::FileChanged) {
+                                    eprintln!("Error sending through channel: {e}");
+                                }
+                            }
+                        }
+                    }
+                }
+            })
+            .unwrap();
+
+        notify_watcher.watch(
+            watcher.root_env.parent().unwrap(),
+            RecursiveMode::NonRecursive,
+        )?;
+        for package_path in watcher.packages.iter() {
+            notify_watcher.watch(package_path, RecursiveMode::Recursive)?;
```

Comment:
It would be nice to be able to filter out the temporary files here. But that is an optimization we can deal with later (though I doubt we'll ever need)

Reply:
I think the difficulty is that its watching the whole directory so that it catches new files, so we couldn't calculate if a new file matches the temporary file pattern a-priori. Maybe there is a method I'm not aware of that does that you propose.

Reply:
Yeah no worries. Again it's filtering later works just as well.